Packages/com.unity.inputsystem/InputSystem/Actions/InputActionState.cs

```diff
@@ -4139,6 +4143,19 @@ public struct UnmanagedMemory : IDisposable
 
             public ActionMapIndices* mapIndices;
 
+            // This is Anthony's version of block allocation from a blob
+            private byte* AllocFromBlob(ref byte* top, int size)
+            {
+                if (size == 0)
+                    return null;
```

Comment:
The largest diff here is this size check, which would guarantee that a zero-size array would be null by design. Priorly zero-sized data would cause invalid non-null pointers. So code that wasn't accurately written to check for bounds would access random memory. This way we might get some crashes more apparently, which is good as otherwise it's super tricky to debug code that uses garbage.

Reply:
Makes sense to have it in a function like this, looks like it could even be static?

Reply:
Might it even be a local function within Allocate?